src/node/server/serverPluginModuleRewrite.ts

```diff
@@ -300,3 +326,72 @@ export const resolveImport = (
   }
   return id
 }
+
+const analysisCache = new Map<
+  string,
+  { isCommonjs: { [name: string]: true } }
+>()
+
+/**
+ * get analysis result from optimize step:
+ * which optimized dependencies may be commonjs
+ */
+function getAnalysis(root: string): { isCommonjs: { [name: string]: true } } {
+  if (analysisCache.has(root)) return analysisCache.get(root)!
+  const cacheDir = resolveOptimizedCacheDir(root)
+  if (!cacheDir) throw new Error('cacheDir not found')
+  const analysis = fs.readJsonSync(path.join(cacheDir, '_analysis.json'))
```

Comment:
You should first check if the file exists and maybe catch errors if json is invalid

Reply:
and return a default value if file does not exist

Reply:
Can you wrap this code in a try catch? it will fail if the user has set optimizeDeps.auto to false

Reply:
Done! Thanks!